apis/r/src/Makevars.in

```diff
@@ -17,7 +18,13 @@ all: $(SHLIB)
 	if [ `uname -s` = 'Darwin' ] && [ -f tiledbsoma.so ]; then \
 	    if [ -f ../inst/tiledb/lib/libtiledb.dylib ] ; then \
 	        install_name_tool -change libz.1.dylib @rpath/libz.1.dylib ../inst/tiledb/lib/libtiledb.dylib; \
-		install_name_tool -add_rpath @loader_path/../tiledb/lib tiledbsoma.so; \
-	    	install_name_tool -add_rpath @loader_path/../tiledbsoma/lib tiledbsoma.so; \
+	        install_name_tool -add_rpath @loader_path/../tiledb/lib tiledbsoma.so; \
+	        install_name_tool -add_rpath @loader_path/../tiledbsoma/lib tiledbsoma.so; \
+	    fi; \
+	    if [ -f /usr/local/lib/libtiledb.dylib -o -f /usr/local/lib/libtiledbsoma.dylib ] ; then \
+	        install_name_tool -add_rpath /usr/local/lib tiledbsoma.so; \
 	    fi; \
 	fi
+
+# TODO: the /usr/local check is a temporary heuristic which is OK for now.
+# Longer-term work is tracked on https://github.com/single-cell-data/TileDB-SOMA/issues/1365
```

Comment:
These two lines do not belong here.  It is understand "everything is in flux" as we plan to get binary library artificacts too. 

The 'macOS allows local compilation but it opinionated enough to not have `pkg-config` is really a root issue here.  We could just follow CRAN and enforce presence of `pkg-config` in the local compilation case.  It does not strike me as that outrageous a demand, and the .pc file we create with `cmake` do work everywhere else.